### DISCUSSION TITLE
use GitHub pages for adding a documentation

### DISCUSSION THREAD

**Discussion by 0xMRTT** [topgrade-rs/topgrade#5] [Ideas]
I would suggest to use mdbook and GitHub pages for adding documentation.

So, user will just be able to go to the documentation website instead of the wiki. Maybe a repo in the org for a short URL. I'll glad to work on that if you approve

**Comment by s34m** [comment]
On one side we could improve the documentation with using mdbook, but i like the idea of having a docs.rs page more becauseit makes the maintainability easier because of in-code documentation and also will allow us in the future to split of the backend from the frontend and publishing the backend in an own lib crate.

**Comment by 0xMRTT** [comment]
I was thinking as using mdbook as a website for configuration without need to now all internal stuff, just information about usage and configuration

**Comment by s34m** [comment]
Sounds good for me.

**Comment by s34m** [comment]
I would go with a dedicated repository then. What do you need for that? New repo with read/write rights?

**Comment by 0xMRTT** [comment]
Yes, a repo called `topgrade-rs.github.io` with github workflow, pages, commit acceess

**Comment by s34m** [comment]
I set you as a maintainer of the new repository. It should work for now.

Edit: Typo

**Comment by GideonBear** [comment]
Since the website now exists, I'll close this. Any further communication about the website should take place in https://github.com/topgrade-rs/topgrade-rs.github.io.
